tests/test/wf_lowering.rs

```diff
@@ -1438,3 +1438,21 @@ fn coerce_unsized_struct() {
         }
     }
 }
+
+#[test]
+fn no_discriminant_kind_impls() {
```

Comment:
Can we move this to `discriminant_kind` test file? Would like to start putting related tests together.